### DISCUSSION TITLE
server.send buffer size increase

### DISCUSSION THREAD

**Discussion by jlemieux55** [khoih-prog/WebSockets2_Generic#56] [Q&A]
Hi,
Been using the websocket and webserver libraries, nice work!
I need to increase the buffer size on the server.send command. My html is over 4500bytes.
Found a buffer size setup for "send_p" but not "send".
Is their a workaround? found the "stream" function but not sure what it does.

Thanks

**Comment by khoih-prog** [comment]
Hi @jlemieux55 

You have to post the [**MRE**](https://stackoverflow.com/help/minimal-reproducible-example), so that anybody can duplicate the issue, and knows that the library needs some **enhancement**, not a bug in your code.

If you can do, I suggest you post the issue on relevant place, such as [**WiFiWebServer**](https://github.com/khoih-prog/WiFiWebServer), [**EthernetWebServer**](https://github.com/khoih-prog/EthernetWebServer) library, etc.

If this is a **valid request**, I'll add functions similar to `send_P()` to permit you to change the buffer as well as **sending in chunks** to deal with extremely large HTMLs.

Just in case if you're interested in Async, check my recent additions of many `SendChunked` examples in there AsyncWebServer libaries

1. [**AsyncWebServer_WT32_ETH01**](https://github.com/khoih-prog/AsyncWebServer_WT32_ETH01)
2. [**AsyncWebServer_RP2040W**](https://github.com/khoih-prog/AsyncWebServer_RP2040W)
3. [**AsyncWebServer_Ethernet**](https://github.com/khoih-prog/AsyncWebServer_Ethernet)
4. [**AsyncWebServer_Teensy41**](https://github.com/khoih-prog/AsyncWebServer_Teensy41)
5. [**Portenta_H7_AsyncWebServer**](https://github.com/khoih-prog/Portenta_H7_AsyncWebServer)
6. [**AsyncWebServer_STM32**](https://github.com/khoih-prog/AsyncWebServer_STM32)

such as

[**AsyncWebServer_SendChunked**](https://github.com/khoih-prog/AsyncWebServer_RP2040W/tree/main/examples/AsyncWebServer_SendChunked) example.
